modules/database/database.bal

```diff
@@ -1456,5 +1459,9 @@ public isolated function timelineStepListFilter(int experimentId, string? plugin
             filter = sql:queryConcat(filter, ` TimelineSubstep.cleared = 1) `);
         }
     }
+    if result\-quality != () && result\-quality != "" {
+        string resultQualityString = "%" + result\-quality + "%";
+        filter = sql:queryConcat(filter, ` AND resultQuality LIKE ${resultQualityString} `);
```

Comment:
result quality can and should use exact matching semantics without `%` (i.e., use `=` instead of `LIKE` for this comparison)